Controller/Heroes/Baccarat/CharacterCards/AceOfSwordsSwordsBaccaratCharacterCardController.cs

```diff
@@ -0,0 +1,158 @@
+                        {
+                            base.GameController.ExhaustCoroutine(coroutine3);
+                        }
+                        break;
+                    }
+            }
+            yield break;
+        }
+        private IEnumerator DealDamageAndDrawResponse(Card card)
+        {
+            if (card != null)
+            {
+                IEnumerator coroutine = base.DealDamage(card, card, 1, DamageType.Toxic, false, false, false, null, null, null, false, null);
+                IEnumerator coroutine2 = base.DrawCard(card.Owner.ToHero());
+                if (base.UseUnityCoroutines)
+                {
+                    yield return base.GameController.StartCoroutine(coroutine);
+                    yield return base.GameController.StartCoroutine(coroutine2);
+                }
+                else
+                {
+                    base.GameController.ExhaustCoroutine(coroutine);
+                    base.GameController.ExhaustCoroutine(coroutine2);
+                }
+                this.LogActedCard(card);
+                coroutine2 = null;
+            }
+            yield break;
+        }
+
+        private void LogActedCard(Card card)
+        {
+            if (card.SharedIdentifier != null)
+            {
+                IEnumerable<Card> collection = base.FindCardsWhere((Card c) => c.SharedIdentifier != null && c.SharedIdentifier == card.SharedIdentifier && c != card, false, null, false);
+                this.actedHeroes.AddRange(collection);
+
+            }
+        }
+
+        public override IEnumerator UsePower(int index = 0)
+        {
+            //Play X copies of Afterlife Euchre from your trash (up to 3). Discard the top 3 - X cards of your deck.
+            int X = 0;
+            bool skipped = false;
+            List<SelectCardDecision> decisions = new List<SelectCardDecision>();
+            Card selectedCard;
+            List<Card> playedCards = new List<Card>();
+            int upTo = GetPowerNumeral(0, 3);
+            int discardBase = GetPowerNumeral(1, 3);
+            IEnumerator coroutine;
+            while (X < upTo && !skipped)
+            {
+                //Play X copies of Afterlife Euchre from your trash (up to 3).
+                coroutine = base.GameController.SelectAndMoveCard(base.HeroTurnTakerController, (Card c) => c.Identifier == "AfterlifeEuchre" && !playedCards.Contains(c) && c.Location == base.TurnTaker.Trash, base.TurnTaker.PlayArea, optional: true, playIfMovingToPlayArea: true, storedResults: decisions, cardSource: base.GetCardSource());
+                if (base.UseUnityCoroutines)
+                {
+                    yield return base.GameController.StartCoroutine(coroutine);
+                }
+                else
+                {
+                    base.GameController.ExhaustCoroutine(coroutine);
+                }
+                selectedCard = decisions.LastOrDefault<SelectCardDecision>().SelectedCard;
```

Comment:
Prefer DidSelectCard() and GetSelectedCard() built-ins

Reply:
DidSelectCard() returns false before first pass, causing the power to not trigger,
GetSelectedCard() returns the last selected card, it ignores skips/nulls